system-contracts/contracts/ContractDeployer.sol

```diff
@@ -284,7 +345,14 @@ contract ContractDeployer is IContractDeployer, SystemContractBase {
             revert NonEmptyAccount();
         }
 
-        _performDeployOnAddress(_bytecodeHash, _newAddress, _aaVersion, _input);
+        _performDeployOnAddress(_bytecodeHash, _newAddress, _aaVersion, _input, true);
+    }
+
+    function _evmDeployOnAddress(address _newAddress, bytes calldata _initCode) internal {
+        // Unfortunately we can not provide revert reason as it would break EVM compatibility
+        // TODO: maybe it is redundant
```

Comment:
afaik, it is not redundant. Even more, we need to also check that nonce is zero

https://eips.ethereum.org/EIPS/eip-684
https://eips.ethereum.org/EIPS/eip-1014

Reply:
Right, I added the proper check and removed the TODO comment.